### ISSUE TITLE
default genesis function

### ISSUE THREAD

**Issue by thedavidmeister** [holochain/holochain-proto#657]
an empty zome file should not trigger an error but it does:

```
web: Error during chain genesis: In 'sampleZomeB' zome: Error executing genesis: ReferenceError: 'genesis' is not defined
```

can we provide the simplest possible genesis as a fallback?

```javascript
function genesis () {
  return true;
}
```

**Comment by Connoropolous** [comment]
I'm a little conflicted as to whether it should work like that...

if its not required that it be defined then people may not know that it should exist, or what it does.

**Comment by thedavidmeister** [comment]
@Connoropolous let's compare some examples of other platforms and see what's "normal"

**Comment by Connoropolous** [comment]
Thinking of npm package.json files makes me realize that their pattern could be a good compromise, which is displaying a warning when running CLI commands. Something simple like "genesis function is not defined in zome code file".

**Comment by Connoropolous** [comment]
So it will still let you run it, and you can ignore it, but it also prompts you to think about adding it, because you may realize there's a reason to add it

**Comment by thedavidmeister** [comment]
@Connoropolous the thing is that there are lots of other api methods that also aren't set, why treat genesis differently?

**Comment by thedavidmeister** [comment]
> if its not required that it be defined then people may not know that it should exist, or what it does.

this is _all_ the current API functions right now :/

**Comment by Connoropolous** [comment]
Because genesis gets run in every single instance, regardless of if you have no other app code.

**Comment by Connoropolous** [comment]
That's not true for the rest of them. Holochain *will* throw an error if it tries to trigger a hook to any of the other functions that it needs but you haven't defined. 

For example, if you try to use `commit` without defining `validateCommit` it will do the same thing. `genesis` just happens to have no dependencies on other functions, it gets called every time

**Comment by thedavidmeister** [comment]
why?

**Comment by thedavidmeister** [comment]
> Holochain will throw an error if it tries to trigger a hook to any of the other functions that it needs but you haven't defined.

hmmm ok, i'll need to think more about that to form an opinion

**Comment by Connoropolous** [comment]
I guess that it is a pattern inherited from blockchain, where the concept of genesis is quite important. The idea is that each zome should specify its' `genesis` behaviour.

**Comment by thedavidmeister** [comment]
right, but "hooks" typically are optional for injecting/overriding some custom behaviour only where needed...

making the fns required but then only throwing errors ad-hoc at runtime seems to me to be asking for trouble when it comes time to compose things arbitrarily, but i'll think about this a bit more

**Comment by Connoropolous** [comment]
Yeah... it would be nice if we could do some checking within the file at first execution or something. Maybe return an error if you use `commit` but don't define `validateCommit`.

**Comment by thedavidmeister** [comment]
yeah, moving the errors to compile time would help avoid edge case errors for sure, although it still seems like a lot of boilerplate to get the value `true` to allow the function you just called to run.

if the goal is just to let users know these functions exist, is enforcing boilerplate the best/only way to achieve that?

**Comment by thedavidmeister** [comment]
expanding the capabilities and quality of the boilerplate generated might be an alternative route here too - https://github.com/holochain/holochain-proto/issues/662
